cmd/antrea-agent/agent_windows.go

```diff
@@ -0,0 +1,98 @@
+package main
+
+import (
+	"os/exec"
+
+	"golang.org/x/sys/windows"
+	"golang.org/x/sys/windows/svc/mgr"
+)
+
+// Initializer knows how to setup host the plugin for self-registration
+type ServiceInitializer struct {
+	serviceNameFlag  string
+}
+
+func newServiceInitializer(
+	serviceNameFlag  string) *ServiceInitializer {
+	return &ServiceInitializer{
+		serviceNameFlag:     serviceNameFlag,
+	}
+}
+
+func getServicePath() (string, error) {
+	p, err := exec.LookPath(os.Args[0])
+	if err != nil {
+		return "", err
+	}
+	return filepath.Abs(p)
+}
+
+// registerService is an entrypoint early in the daemon startup
+// to handle registering against Windows Service Control Manager (SCM).
+func (i *Initializer) registerServiceforOS() error {
+
+	p, err := getServicePath()
+	if err != nil {
+		return err
+	}
+	m, err := mgr.Connect()
+	if err != nil {
+		return err
+	}
+	defer m.Disconnect()
+
+	c := mgr.Config{
+		ServiceType:  windows.SERVICE_WIN32_OWN_PROCESS,
+		StartType:    mgr.StartAutomatic,
+		ErrorControl: mgr.ErrorNormal,
+		DisplayName:  "Antrea",
+		Description:  "Container network plugin",
+	}
+
+	// Configure the service to launch with the arguments that were just passed.
+	args := []string{"--run-service"}
+	for _, a := range os.Args[1:] {
+		if a != "--register-service" && a != "--unregister-service" {
```

Comment:
Where are the flags defined? I think the program will fail to start if the flags are specified but not defined. And I think you may want to check if "--register-service" is set before registering the service.